### PR TITLE
FAQs and Known Implementations

### PR DESCRIPTION
**Pull request by DirectXMan12**
This adds some FAQs to the readme, plus corrects the list of known implementations to not just be glog.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,140 @@ may feel very similar, but the primary difference is the lack of semantics.
...
+For instance, consider the following examples (all taken from spots in the
+Kubernetes codebase):
+
+- `klog.V(4).Infof("Client is returning errors: code %v, error %v",
+  responseCode, err)` becomes `logger.V(4).Error(error, "client returned an
+  error", "code", responseCode)
+
+- `klog.V(4).Infof("Got a Retry-After %ds response for attempt %d to %v",
+  seconds, retries, url)` becomes `logger.V(4).Info("got a retry-after
+  response when requesting url", "attempt", retries, "after
```
Comment (thockin) [inline README.md]: Should we require or suggest that keys be "tokens" (no whitespace) or do we not care?  Should we spec a suggested character set, but not require?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,140 @@ may feel very similar, but the primary difference is the lack of semantics.
...
+  seconds", seconds, "url", url)`
+
```
Comment (thockin) [inline README.md]: If you REALLY want formatted strings, you can `fmt.Sprintf` it yourself.

Comment (DirectXMan12) [reply README.md]: I'll clarify.  I mostly meant "don't use a fmt.Sprintf" message.

File: README.md (hunk 1 above)
Comment (DirectXMan12) [reply README.md]: I thus far haven't required tokens (I actively use the opposite).  I haven't encountered a case where it's an issue so far.  Suggesting a character set seems reasonable.

**Comment by DirectXMan12** [comment]
comments addressed

**Pull request merged**